object: Allow header with payload chunk in first GET response message

cthulhu-rider opened the pull request:
here's an option through the proto version. It will work on the wire. The only other alternative I see is a new `GetV2` call which is more demanding of support

cthulhu-rider commented:
https://github.com/nspcc-dev/neofs-node/issues/3396#issuecomment-2980001356

carpawell reviewed (commented):
Do we have any load updates on it?

cthulhu-rider committed:
object: Allow header with payload chunk in first GET response message

Previously, heading and chunk fields were mutually exclusive in each GET
response message. With this, server sent 1st message with the header
only, then messages with payload chunks came. This behavior is
inefficient since any object with a non-empty payload requires one more
message. The smaller the payload, the greater the overhead.

This modifies GET protocol to allow header and chunk to be attached to
the same message. For backward compatibility, requests from older
clients will be processed as before.

Signed-off-by: Leonard Lyubich <leonard@morphbits.io>
Signed-off-by: Leonard Liubich <leonard@morphbits.io>

cthulhu-rider commented:
since new field is needed anyway, decided to just extend heading message. Thus, initial message can carry entire object completely
